src/client/content_cao.cpp

```diff
@@ -915,6 +915,30 @@ void GenericCAO::updateNodePos()
 	}
 }
 
+void GenericCAO::pushMeshAnimFrame(float dtime)
+{
+	if (m_animated_meshnode) {
+		// Compensate for the time lost remaking the scene node.
+		// Not an exact fix - the time may be a tiny bit ahead.
+		s32 delta = (s32)(dtime / m_animated_meshnode->getAnimationSpeed() + 1);
```

Comment:
When I applied the fix without this (simply stashing and restoring the frame), repeatedly changing the properties would slow the animation down. I suppose it's because irrlicht gives us the last whole animation frame, which is always behind and not precise enough. By some trial and error, I came up with this to reconstruct the proper time as closely as possible.